File: Assets/My Assets/Scripts/UI/FadeUIElement.js
var fade : boolean = false; //false fade out / true fade in
var fadevalue : float = 0;
var targetAlpha : float = 0.2;
var speed : float = 0.5;
function Update(){
	if(fade == true && fadevalue < targetAlpha){
		fadevalue+=Time.deltaTime * speed;
	}
	if(fade == false && fadevalue > 0){
		fadevalue-=Time.deltaTime * speed;
	}
	capFade();
	GetComponent("Graphic").color.a = fadevalue;
}
function vignetteToggle(mode : int){
	if(mode == 1){ //action
		fade = true;
	}
	else{ 
		fade = false;
	}
}
function capFade(){
	if(fadevalue > targetAlpha){
		fadevalue = targetAlpha;
	}
	if(fadevalue < 0){
		fadevalue = 0;
	}
}

File: Assets/My Assets/Scripts/UI/FadeUIElement.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class FadeUIElement : MonoBehaviour
{
    public bool fade; //false fade out / true fade in
    public float fadevalue;
    public float targetAlpha;
    public float speed;
    public virtual void Update()
    {
        if ((this.fade == true) && (this.fadevalue < this.targetAlpha))
        {
            this.fadevalue = this.fadevalue + (Time.deltaTime * this.speed);
        }
        if ((this.fade == false) && (this.fadevalue > 0))
        {
            this.fadevalue = this.fadevalue - (Time.deltaTime * this.speed);
        }
        this.capFade();
        this.GetComponent("Graphic").color.a = this.fadevalue;
    }

    public virtual void vignetteToggle(int mode)
    {
        if (mode == 1) //action
        {
            this.fade = true;
        }
        else
        {
            this.fade = false;
        }
    }

    public virtual void capFade()
    {
        if (this.fadevalue > this.targetAlpha)
        {
            this.fadevalue = this.targetAlpha;
        }
        if (this.fadevalue < 0)
        {
            this.fadevalue = 0;
        }
    }

    public FadeUIElement()
    {
        this.targetAlpha = 0.2f;
        this.speed = 0.5f;
    }

}